rules/autogen.bzl

```diff
@@ -27,9 +27,35 @@ def _autogen_hjson_header(ctx):
     ))]
 
 autogen_hjson_header = rule(
-    implementation = _autogen_hjson_header,
+    implementation = _hjson_header,
     attrs = {
         "srcs": attr.label_list(allow_files = True),
         "_tool": attr.label(default = "//util:regtool.py", allow_files = True),
     },
 )
+
+def _chip_info(ctx):
+    header = ctx.actions.declare_file("chip_info.h")
+    ctx.actions.run(
+        outputs = [header],
+        inputs = ctx.files.version + ctx.files._tool,
+        arguments = [
+            "-o",
+            header.dirname,
+            "--ot_version_file",
+            ctx.files.version[0].path,
+        ],
+        executable = ctx.files._tool[0],
+    )
+    return [CcInfo(compilation_context = cc_common.create_compilation_context(
+        includes = depset([header.dirname]),
+        headers = depset([header]),
+    ))]
```

Comment:
As a general rule, we should make all of our autogen rules emit both a CcInfo provider and a DefaultInfo provider.

The latter is generally useful if you just want to type `bazel build :chip_info` and see the produced file (the CcInfo provider appears to not do this by default - only when it is consumed by another cc_* rule.  This omission was an oversight on my part).

Reply:
That makes sense